### PR TITLE
Frontend system > Update user API integration

### PR DESCRIPTION
**Pull request by salman-codewar**
## Purpose changes

**Registry - API GW**
- Added Deactivate user endpoint
- Added Delete user endpoint
 
**Frontend system**
- Updated AddUser mutation
- Updated UpdateUser mutation
- Updated DeactivateUser mutation
- Updated DeleteUser mutation
- Updated GetUser query

### CODE REVIEW

File: systems/registry/api-gateway/pkg/client/users.go [hunk 1]
```diff
@@ -47,10 +47,21 @@ func (r *Users) Get(userUUID string, requesterId string) (*pbusers.GetResponse,
 	return r.client.Get(ctx, &pbusers.GetRequest{UserUuid: userUUID})
 }
 
-func (r *Users) AddUser(user *pbusers.User, requesterId string) (*pbusers.AddResponse, error) {
+func (r *Users) Update(userUUID string, user *pbusers.UserAttributes, requesterId string) (*pbusers.UpdateResponse, error) {
```
Comment (Brackleycassinga) [inline systems/registry/api-gateway/pkg/client/users.go:50]: Can we rename `useruuid` to `userId` for the sake of readability?

File: systems/registry/api-gateway/pkg/client/users.go [hunk 2]
```diff
@@ -47,10 +47,21 @@ func (r *Users) Get(userUUID string, requesterId string) (*pbusers.GetResponse,
...
 	ctx, cancel := r.getContext(requesterId)
 	defer cancel()
+	return r.client.Update(ctx, &pbusers.UpdateRequest{UserUuid: userUUID, User: user})
+}
+
+func (r *Users) Deactivate(userUUID string, requesterId string) (*pbusers.DeactivateResponse, error) {
```
Comment (Brackleycassinga) [inline systems/registry/api-gateway/pkg/client/users.go:56]: Can we rename `useruuid` to `userId` for the sake of readability?

File: systems/registry/api-gateway/pkg/client/users.go [hunk 3]
```diff
@@ -68,20 +79,11 @@ func (r *Users) UpdateUser(userUUID string, user *pbusers.UserAttributes, reques
...
-func (r *Users) Delete(userUUID string, requesterId string) error {
-	ctx, cancel := r.getContext(requesterId)
-	defer cancel()
-
-	_, err := r.client.Delete(ctx, &pbusers.DeleteRequest{UserUuid: userUUID})
-	return err
-}
-
-func (r *Users) DeactivateUser(userUUID string, requesterId string) error {
+func (r *Users) Delete(userUUID string, requesterId string) (*pbusers.DeleteResponse, error) {
```
Comment (Brackleycassinga) [inline systems/registry/api-gateway/pkg/client/users.go:82]: Can we rename `useruuid` to `userId` for the sake of readability?

**Review by Brackleycassinga — CHANGES_REQUESTED** [review]
Looks good 👍🏾 .I have added some comments.

File: systems/bff/src/modules/user/service.ts [hunk 4]
```diff
@@ -35,57 +33,55 @@ export class UserService implements IUserService {
...
+            body: { name: req.name, email: req.email, phone: req.phone },
         });
         if (checkError(res)) throw new Error(res.message);
         return UserMapper.dtoToUserResDto(res);
     };
     deactivateUser = async (
-        id: string,
-        cookie: ParsedCookie
-    ): Promise<DeactivateResponse> => {
+        uuid: string,
```
[the remark this answers is not in the document]
Comment (salman-codewar) [reply systems/bff/src/modules/user/service.ts:58]: addressed

File: systems/registry/api-gateway/pkg/client/users.go (hunk 3 above)
Comment (salman-codewar) [reply systems/registry/api-gateway/pkg/client/users.go:82]: addressed

Comment (salman-codewar) [reply systems/registry/api-gateway/pkg/client/users.go:82]: Yea, your point is valid. but we're calling it `userUUID` in all other endpoints before. So i just followed the conventions. For that readability improvement i think we should address it separately in registry api-gw.

File: systems/bff/src/modules/user/resolver/getUser.resolver.ts [hunk 5]
```diff
@@ -1,22 +1,22 @@
...
     constructor(private readonly userService: UserService) {}
 
-    @Query(() => GetUserDto)
+    @Query(() => UserResDto)
     @UseMiddleware(Authentication)
     async getUser(
         @Arg("userId") userId: string,
-        @Ctx() ctx: Context
-    ): Promise<GetUserDto | null> {
+        @Ctx() ctx: Context,
```
[the remark this answers is not in the document]
Comment (salman-codewar) [reply systems/bff/src/modules/user/resolver/getUser.resolver.ts:18]: I've updated the eslint rules with `tailingComma` so we have to bear these changes once.

**Pull request merged**